app/src/main/java/org/simple/clinic/widgets/PatientSearchResultItemView.kt

```diff
@@ -93,6 +107,48 @@ class PatientSearchResultItemView(
     renderPatientPhoneNumber(searchQuery, model)
     renderVisited(model.lastSeen)
     renderLastSeen(model.lastSeen, currentFacilityId)
+    renderIdentifier(model.identifier, searchQuery)
+  }
+
+  private fun getId(identifier: Identifier, searchQuery: String): Id {
+    val id = when (identifier.type) {
+      BpPassport -> BpPassport.shortCode(identifier)
+      BangladeshNationalId -> identifier.value
+      EthiopiaMedicalRecordNumber -> identifier.value
+      IndiaNationalHealthId -> identifier.value
+      is Unknown -> throw IllegalArgumentException("Unknown identifier: $identifier")
+    }
+    val canHighlightIdentifier = id.contains(searchQuery)
+
+    return if (canHighlightIdentifier) {
+      val indexOfIdentifier = id.indexOf(searchQuery)
+      Id.Highlighted(
+          value = id,
+          highlightStart = indexOfIdentifier,
+          highlightEnd = indexOfIdentifier + searchQuery.length
+      )
+    } else {
+      Id.Plain(value = id)
+    }
+  }
+
+  private fun renderIdentifier(identifier: Identifier?, searchQuery: String?) {
+    if (identifier == null || searchQuery == null) {
+      identifierContainer.visibility = View.GONE
+    } else {
+      identifierContainer.visibility = View.VISIBLE
+      val id = when (val id = getId(identifier, searchQuery)) {
+        is Id.Highlighted -> {
+          val highlightId = SpannableStringBuilder(id.value)
+          val highlightColor = context.resolveColor(colorRes = R.color.search_query_highlight)
+          highlightId.setSpan(BackgroundColorSpan(highlightColor), id.highlightStart, id.highlightEnd, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)
```

Comment:
What do you think of extracting the construction of the highlight span to a separate method for better readability?

```kotlin
val highlightId = highlight(id.value, id.highlightStart, id,highlightEnd, R.color.search_query_highlight)
```

If we move ahead with my earlier suggestion to use composition over inheritance for representing the search result types, we can even make this a method of the `Highlighted` type.

```kotlin
val highlightId = id.createHighlightedSpan(R.color.search_query_highlight)
```